ompi/mca/common/ompio/common_ompio_file_write.c

```diff
@@ -53,22 +57,64 @@ int mca_common_ompio_file_write (mca_io_ompio_file_t *fh,
     ssize_t ret_code=0;
     size_t spc=0;
     int i = 0; /* index into the decoded iovec of the buffer */
-    int j = 0; /* index into the file view iovec */
+    int j = 0; /* index into the file view iovec */ 
 
     if ( 0 == count ) {
-	if ( MPI_STATUS_IGNORE != status ) {
+        if ( MPI_STATUS_IGNORE != status ) {
 	    status->_ucount = 0;
 	}
 	return ret;
     }
 
+#if OPAL_CUDA_SUPPORT
+    int is_gpu=0, is_managed=0;
```

Comment:
This part of the code (the convertor preparation, buffer registration and iovec allocation) seems similar with the one above (with the exception of the pack). Moving them outside in macros/functions will simplify and consolidate the code.

Reply:
done